### PR TITLE
Sticky container titles

### PR DESCRIPTION
**Pull request by stephanfowler**
A zero-audience AB test. Turn on by adding: `#ab-StickyContainerTitles=show`

sticky-bottom-titles

Titles are clickable, causing a jump to that container; thereafter they have their normal click behaviour.

### CODE REVIEW

**Comment by michaelwmcnamara** [comment]
Issue list here: https://docs.google.com/a/guardian.co.uk/spreadsheets/d/1MKvEoemEsV7brjoVL2MOx_y1XJdzj3JIsaGzP3ztWOY/edit#gid=0

**Comment by phamann** [comment]
@stephanfowler Looks nice! 

I'd prefer it if this uses native `position: sticky` if supported and then polyfills if not. We already have a module written for this [here](https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/ui/sticky.js)

**Comment by ironsidevsquincy** [comment]
note that module doesn't use the native `position: sticky` if available - see todo. but can be made to, if anyone has time...

File: static/src/javascripts/projects/common/modules/experiments/tests/sticky-container-titles.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+        this.variants = [
+            {
+                id: 'control',
+                test: function () { }
+            },
+            {
+                id: 'show',
+                test: function () {
+                    if (['Network Front', 'Section'].indexOf(config.page.contentType) > -1) {
```
Comment (ironsidevsquincy) [inline static/src/javascripts/projects/common/modules/experiments/tests/sticky-container-titles.js]: no IE8 support for `indexOf` - https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf#Browser_compatibility

File: static/src/stylesheets/module/facia/_container.scss [hunk 2]
```diff
@@ -142,6 +143,59 @@ $header-image-size-desktop: 100px;
     }
 }
 
+.fc-container__header__title--sticky {
+    display: block;
+    width: 100%;
+    text-align: left;
+    font-weight: normal;
+    margin-left: $gs-gutter * -1;
+    padding: $gs-gutter/4 0 0 $gs-gutter;
```
Comment (ironsidevsquincy) [inline static/src/stylesheets/module/facia/_container.scss]: `gs-basline` for vertical spacing

File: static/src/stylesheets/module/facia/_container.scss [hunk 3]
```diff
@@ -142,6 +143,59 @@ $header-image-size-desktop: 100px;
...
+        cursor: pointer;
+    }
+
+    &.onscreen {
+        @include f-header();
+    }
+}
+
+.fc-container__header__title--sticky:last-child {
+    padding-bottom: $gs-gutter;
```
Comment (ironsidevsquincy) [inline static/src/stylesheets/module/facia/_container.scss]: gs-basline for vertical spacing

**Comment by stephanfowler** [comment]
@phamann I'm not convinced `position: sticky` is worth using/polyfiling. Support is really poor and a general pollyfill is kinda complex. Also, the sticky.js module is for sticking to top, whereas this PR sticks to bottom and in pretty non-generalisable way. So I'd prefer to leave this as it is...

**Comment by phamann** [comment]
Fair do's, both valid arguments. It's also just an experiment :wink:

**Pull request merged**